### PR TITLE
Layout native test components correctly on unix

### PR DESCRIPTION
**Pull request by jashook**
Small tweaks to build-test.sh. This will enable the work that @AaronRobinsonMSFT did by default. In addition, it removes some unused components of build-test.sh (BuildAgainstPackages) and adds skipmanaged to the option list.

In addition, it finishes laying out the native test components next to the managed test when building tests using build-test.sh.

### CODE REVIEW

File: build-test.sh [hunk 1]
```diff
@@ -270,24 +266,26 @@ build_Tests()
...
-    if [ $? -ne 0 ]; then
-        echo "${__MsgPrefix}Error: build failed. Refer to the build log files for details (above)"
-        exit 1
-    else
-        echo "Checking the Managed Tests Build..."
+    if [ $__SkipManaged != 1 ]; then
+        echo "Starting the Managed Tests Build..."
 
-        build_MSBuild_projects "Check_Test_Build" "${__ProjectDir}/tests/runtest.proj" "Check Test Build" "-ExtraParameters:/t:CheckTestBuild"
+        build_MSBuild_projects "Tests_Managed" "$__ProjectDir/tests/build.proj" "Managed tests build (build tests)" "$__up"
```
Comment (AaronRobinsonMSFT) [inline build-test.sh]: Please keep braces for variable names (i.e. `${__ProjectDir}`) in bash.

Comment (jashook) [reply build-test.sh]: This is a whitespace diff, only has been indented.

Comment (AaronRobinsonMSFT) [reply build-test.sh]: Left side:

> `build_MSBuild_projects "Check_Test_Build" "${__ProjectDir}/tests/runtest.proj" "Check Test Build" "-ExtraParameters:/t:CheckTestBuild"`

Right side:

> `build_MSBuild_projects "Tests_Managed" "$__ProjectDir/tests/build.proj" "Managed tests build (build tests)" "$__up"`

The above is what I am seeing on github. If your changes are not removing the braces around `__ProjectDir`, then that is fine, but in general we should not be removing braces from environment variable access that are embedded in strings.

Comment (jashook) [reply build-test.sh]: Correct, this variable already was missing the braces. None have been removed. We should most likely add some; however, I will consider that out of scope for this change.

File: build-test.sh [hunk 2]
```diff
@@ -257,10 +257,6 @@ build_Tests()
         build_MSBuild_projects "Restore_Product" "${__ProjectDir}/tests/build.proj" "Restore product binaries (build tests)" "-BatchRestorePackages"
     fi
 
-    if [ -n "$__BuildAgainstPackagesArg" ]; then
-        build_MSBuild_projects "Tests_GenerateRuntimeLayout" "${__ProjectDir}/tests/runtest.proj" "Restore product binaries (run tests)" "-BinPlaceRef" "-BinPlaceProduct" "-CopyCrossgenToProduct"
-    fi
-
```
Comment (sbomer) [inline build-test.sh]: Since we're removing this, could you get rid of the buildagainstpackages argument?

**Comment by jashook** [comment]
@dotnet-bot help

File: build-test.sh (hunk 2 above)
Comment (jashook) [reply build-test.sh]: Yes thank you I missed this.

Comment (jashook) [reply build-test.sh]: Addressed

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test
@dotnet-bot test Windows_NT x64 Checked Build and Test
@dotnet-bot test Windows_NT x64 Release Build and Test
@dotnet-bot test Windows_NT x86 Debug Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test
@dotnet-bot test Windows_NT x86 Release Build and Test
@dotnet-bot test Windows_NT arm Cross Debug normal Build
@dotnet-bot test Windows_NT arm Cross Checked normal Build
@dotnet-bot test Windows_NT arm Cross Release normal Build
@dotnet-bot test Windows_NT arm64 Cross Debug normal Build
@dotnet-bot test Windows_NT arm64 Cross Checked normal Build
@dotnet-bot test Windows_NT arm64 Cross Release normal Build
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by jashook** [comment]
Blocked by #19951

**Comment by jashook** [comment]
Failures are unrelated.

**Comment by Chrisboh** [comment]
Hi @jashook, it looks like the machine has already been cleaned up.  Let me know if you start seeing this again.

**Pull request merged**